Client/Frontend/Home/RemoteTabsPanel.swift

```diff
@@ -82,28 +84,46 @@ class RemoteTabsPanel: UITableViewController, HomePanel {
             return
         }
 
-        // Otherwise, fetch the tabs cloud
-        self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
-            self.refreshControl?.endRefreshing()
-
+        self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
             if let clientAndTabs = result.successValue {
-                if clientAndTabs.count == 0 {
-                    self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .NoClients)
-                } else {
-                    let nonEmptyClientAndTabs = clientAndTabs.filter { $0.tabs.count > 0 }
-                    if nonEmptyClientAndTabs.count == 0 {
-                        self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .NoTabs)
-                    } else {
-                        self.tableViewDelegate = RemoteTabsPanelClientAndTabsDataSource(homePanel: self, clientAndTabs: nonEmptyClientAndTabs)
-                        self.tableView.allowsSelection = true
+                self.updateDelegateClientAndTabData(clientAndTabs)
+            } else {
+                self.tableViewDelegate = RemoteTabsPanelErrorDataSource(homePanel: self, error: .FailedToSync)
```

Comment:
This bit is my only concern — if we fail to get cached tabs for some reason, we'll flicker an error view until the remote ones load. Maybe that's no big deal. Otherwise, LGTM.